transactions/transactor.go

```diff
@@ -112,7 +112,9 @@ func (t *Transactor) NextNonce(ctx context.Context, rpcClient rpc.ClientInterfac
 	// We need to take into consideration all pending transactions in case of Optimism, cause the network returns always
 	// the nonce of last executed tx + 1 for the next nonce value.
 	if chainID == wallet_common.OptimismMainnet ||
-		chainID == wallet_common.OptimismSepolia {
+		chainID == wallet_common.OptimismSepolia ||
+		chainID == wallet_common.BaseMainnet ||
```

Comment:
hmm @saledjenic any idea about this? If this is correct, please update the comment since it only mentions Optimism.

Reply:
Was wandering the same, I hope Brian tested it, cause no other way to figure out this.

Reply:
But on the other side, I don't know if we need this at all. I mean, I added that condition a long time ago, but recently (a few months ago) I've faced unreliable nonce resolving on nodefleet for Optimism due to not synced nodes. Since the effect is the same, now I am leaning more toward that we don't need this check. Maybe we should try removing this block entirely and thoroughly inspect sending on Optimism.
@alaibe wdyt?

Reply:
Base works the same as Optimism as it is based on the Optimism stack, so if it is a network behavior we should replicate on Base as well. I can update the comment for now, in case this logic is not needed anymore we can remove it in a follow-up PR.

Reply:
Updated